sorting.go

```diff
@@ -0,0 +1,205 @@
+//go:build go1.18
+
+package parquet
+
+import (
+	"bytes"
+	"io"
+	"sort"
+)
+
+// SortingWriter is a type similar to GenericWriter but it ensures that rows
+// are sorted according to the sorting columns configured on the writer.
```

Comment:
Somehow I thought the GenericWriter was doing that already. Otherwise what's the point of passing SortingColumns to it ? If that's not the case we might want to add something about it in the GenericWriter doc :)